Source/Laser/LaserProfilesImpl/LaserProfileFromFile.cpp

```diff
@@ -0,0 +1,568 @@
+        m_params.nx = extent[2];
+        if(m_params.nt <= 1) Abort("nt in lasy file must be >=2");
+        if(m_params.nx <= 1) Abort("nx in lasy file must be >=2");
+        if(m_params.ny <= 1) Abort("ny in lasy file must be >=2 in 3D");
+        // Extract grid offset and grid spacing
+        std::vector<double> offset = E.gridGlobalOffset();
+        std::vector<double> position = E_laser.position<double>();
+        std::vector<double> spacing = E.gridSpacing<double>();
+        // Calculate the min and max of the grid
+        m_params.t_min = offset[0] + position[0]*spacing[0];
+        m_params.t_max = m_params.t_min + (m_params.nt-1)*spacing[0];
+        m_params.y_min = offset[1] + position[1]*spacing[1];
+        m_params.y_max = m_params.y_min + (m_params.ny-1)*spacing[1];
+        m_params.x_min = offset[2] + position[2]*spacing[2];
+        m_params.x_max = m_params.x_min + (m_params.nx-1)*spacing[2];
+    }
+#else
+    amrex::ignore_unused(lasy_file_name);
+#endif
+}
+
+void
+WarpXLaserProfiles::FromFileLaserProfile::parse_binary_file(std::string binary_file_name)
+{
+    if(ParallelDescriptor::IOProcessor()){
+        std::ifstream inp(binary_file_name, std::ios::binary);
+        if(!inp) Abort("Failed to open binary file");
+        inp.exceptions(std::ios_base::failbit | std::ios_base::badbit);
+
+        //Uniform grid flag
+        char flag;
+        inp.read(&flag, 1);
+        if(!inp) Abort("Failed to read grid type from binary file");
+
+        //Grid points along t, x and y
+        inp.read(reinterpret_cast<char*>(&m_params.nt), sizeof(uint32_t));
+        inp.read(reinterpret_cast<char*>(&m_params.nx), sizeof(uint32_t));
+        inp.read(reinterpret_cast<char*>(&m_params.ny), sizeof(uint32_t));
+        if(!inp) Abort("Failed to read sizes from binary file");
+        if(m_params.nt <= 1) Abort("nt in binary file must be >=2");
+        if(m_params.nx <= 1) Abort("nx in binary file must be >=2");
+#if (defined(WARPX_DIM_3D) || (defined WARPX_DIM_RZ))
+        if(m_params.ny <= 1) Abort("ny in binary file must be >=2 in 3D");
+#elif defined(WARPX_DIM_XZ)
+        if(m_params.ny != 1) Abort("ny in binary file must be 1 in 2D");
+#endif
+
+        //Coordinates
+        Vector<double> dbuf_t, dbuf_x, dbuf_y;
+        dbuf_t.resize(2);
+        dbuf_x.resize(2);
+#if (defined(WARPX_DIM_3D) || (defined WARPX_DIM_RZ))
+        dbuf_y.resize(2);
+#elif defined(WARPX_DIM_XZ)
+        dbuf_y.resize(1);
+#endif
+        inp.read(reinterpret_cast<char*>(dbuf_t.dataPtr()),
+            dbuf_t.size()*sizeof(double));
+        inp.read(reinterpret_cast<char*>(dbuf_x.dataPtr()),
+            dbuf_x.size()*sizeof(double));
+        inp.read(reinterpret_cast<char*>(dbuf_y.dataPtr()),
+            dbuf_y.size()*sizeof(double));
+        if(!inp) Abort("Failed to read coords from binary file");
+
+        m_params.t_min = static_cast<amrex::Real>(dbuf_t[0]);
+        m_params.t_max = static_cast<amrex::Real>(dbuf_t[1]);
```

Comment:
#if (defined(WARPX_DIM_3D) || (defined WARPX_DIM_RZ))
        m_params.y_min = static_cast<amrex::Real>(dbuf_y[0]);
        m_params.y_max = static_cast<amrex::Real>(dbuf_y[1]);
#endif

Reply:
Works locally, let's wait for the tests. Thank you very much for catching that!